alg/teca_tc_trajectory_scalars.py

```diff
@@ -58,362 +58,356 @@ def set_plot_peak_radius(self, plot_peak_radius):
-                    norm_y[0] = norm_y[0]/nmag
-                    norm_x[npts] = lat_i[npts] - lat_i[npts-1]
-                    norm_y[npts] = -(lon_i[npts] - lon_i[npts-1])
-                    norm_x[npts] = norm_x[npts]/nmag
-                    norm_y[npts] = norm_y[npts]/nmag
-                    # for each wind radius, render a polygon of width 2*wind
-                    # centered on the track. have to break it into continuous
-                    # segments
-                    nwri = len(wind_rad_i)
-                    q = nwri - 1
-                    while q >= 0:
-                        self.plot_wind_rad(lon_i, lat_i, norm_x, norm_y, \
-                            wind_rad_i[q], '-', edge_color if q==0 else red_cmap[q], \
-                            2 if q==0 else 1, red_cmap[q], 0.98, q+4)
-                        q -= 1
-                    # plot the peak radius
-                    if (self.plot_peak_radius):
-                        # peak radius is only valid if one of the other wind radii
-                        # exist, zero out other values
-                        kk = wind_rad_i[0] > 1.0e-6
-                        q = 1
-                        while q < nwri:
-                            kk = np.logical_or(kk, wind_rad_i[q] > 1.0e-6)
-                            q += 1
-                        peak_rad_i[np.logical_not(kk)] = 0.0
-                        self.plot_wind_rad(lon_i, lat_i, norm_x, norm_y, \
-                            peak_rad_i, '--', (0,0,0,0.25), 1, 'none', 1.00, nwri+4)
-                    # mark track
-                    marks = wind_rad_i[0] <= 1.0e-6
-                    q = 1
-                    while q < nwri:
-                        marks = np.logical_and(marks, np.logical_not(wind_rad_i[q] > 1.0e-6))
-                        q += 1
-                    kk = np.where(marks)[0]
-
-                    plt.plot(lon_i[kk], lat_i[kk], '.', linewidth=2, \
-                        color=edge_color,zorder=10)
+        #sys.stderr.write('teca_tc_trajectory_scalars::execute\n')
 
-                    plt.plot(lon_i[kk], lat_i[kk], '.', linewidth=1, \
-                        color='k', zorder=10, markersize=1)
-
-                    marks = wind_rad_i[0] > 1.0e-6
+        import matplotlib.pyplot as plt
+        import matplotlib.patches as plt_mp
+        import matplotlib.image as plt_img
+        import matplotlib.gridspec as plt_gridspec
+
+        # store matplotlib state we modify
+        legend_frame_on_orig = plt.rcParams['legend.frameon']
+
+        # tweak matplotlib slightly
+        plt.rcParams['figure.max_open_warning'] = 0
+        plt.rcParams['legend.frameon'] = 1
+
+        # get the input table
+        in_table = teca_py.as_teca_table(data_in[0])
+        if in_table is None:
+            # TODO if this is part of a parallel pipeline then
+            # only rank 0 should report an error.
+            sys.stderr.write('ERROR: empty input, or not a table\n')
```

Comment:
see previous comment about raising exceptions instead of printing to stderr

Reply:
true. we should raise.